crates/database/src/fee_policies.rs

```diff
@@ -0,0 +1,171 @@
+use {
+    crate::{auction::AuctionId, OrderUid, PgTransaction},
+    sqlx::PgConnection,
+    std::ops::DerefMut,
+};
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct FeePolicy {
+    pub auction_id: AuctionId,
+    pub order_uid: OrderUid,
+    pub kind: FeePolicyKind,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum FeePolicyKind {
+    PriceImprovement {
+        price_improvement_factor: f64,
+        max_volume_factor: f64,
+    },
+    Volume {
+        factor: f64,
+    },
+}
+
+#[derive(Debug, Clone, PartialEq, sqlx::FromRow)]
+struct FeePolicyRow {
+    auction_id: AuctionId,
+    order_uid: OrderUid,
+    kind: FeePolicyKindRow,
+    price_improvement_factor: Option<f64>,
+    volume_factor: Option<f64>,
+}
+
+#[derive(Debug, Clone, PartialEq, sqlx::Type)]
+#[sqlx(type_name = "PolicyKind", rename_all = "lowercase")]
```

Comment:
why lowercae? Didn't this make composite words hard to read? What's our standard naming policy here?

Reply:
We use lowercase for database. Used the same here for consistency.